Support alternate virtualenv directories

Exagone313 opened the pull request:
Fix #10  

EDIT: This could be replaced by a custom environment variable, such as:
```
PYLINT_VENV_DIRECTORIES=.venv:.pyenv:.virtualenv
```

jgosmann reviewed (changes requested):
Thank you! In general this looks good to me. Could you also update `README.rst` (it states the folder names) and prepare an entry in `CHANGES.md`?

Exagone313 commented:
@jgosmann, are you good for the code I made or do you think it would be better to add environment variable support like I suggested?

jgosmann commented:
I'm fine with either. I think the current version is already an improvement, having an environment variable would add some more flexibility on top.

Exagone313 commented:
@jgosmann, I updated it. I named the environment variable `PYLINT_VENV_PATH`. I also updated the files you mentioned.

~~Unrelated to this, I noticed that Pylint 2.14.0 (released a few days ago) broke multiline `init-hook`, so it's no longer possible to run the try-except (as far as I know it can't be made on a single line).~~ EDIT: unsure about this, I couldn't reproduce it anymore.

jgosmann commented on pylint_venv.py:
```diff
@@ -65,11 +65,14 @@ def detect_venv():
-    # Check if a local ".venv" folder exists
+    # Check if a virtualenv directory exists (.venv by default, or alternative
+    # paths given by environment variable)
+    venv_paths = os.getenv("PYLINT_VENV_PATH", ".venv").split(":")
```
Note: This does not support any directories where the name itself contains a colon (`:`). However, I think it is fine because the same is true for Bash's `$PATH` variable and I hope no-one uses colons in the venv directory names (or any other directory names for that matter).

jgosmann commented:
> Unrelated to this, I noticed that Pylint 2.14.0 (released a few days ago) broke multiline init-hook, so it's no longer possible to run the try-except (as far as I know it can't be made on a single line).

I was able to reproduce this and reported the issue upstream as PyCQA/pylint#6888.
